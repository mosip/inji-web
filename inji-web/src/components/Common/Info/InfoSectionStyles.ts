export const InfoSectionStyles = {
    content: "inline-flex flex-col items-center max-w-[370px]",
    title: "text-xl text-center sm:text-2xl font-medium text-gray-800 mb-2",
    message: "text-[var(--iw-color-info)] font-normal text-sm break-words justify-center text-center",
};