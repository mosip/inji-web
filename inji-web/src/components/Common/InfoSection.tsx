import React, {Fragment} from "react";

type InfoSectionProps =
    {
        title?: string,
        actionText?: string
        icon?: React.ReactElement
    }

export function InfoSection({
                                title,
                                actionText,
                                icon
                            }: Readonly<InfoSectionProps>): React.ReactElement {
    return <div
        data-testid={"no-cards-stored"}
        className="bg-white rounded-lg shadow-iw-emptyDocuments p-6 sm:p-8 md:p-10 flex flex-col items-center justify-center min-h-[500px] my-4 sm:my-8 md:my-16"
    >
        {icon && <Fragment>{icon}</Fragment>}
        {title && <h2
            data-testid={"No-Credentials-Title"}
            className="text-xl text-center sm:text-2xl font-medium text-gray-800 mb-2"
        >
            {title}
        </h2>}
        {actionText && <span>{actionText}</span>}
    </div>;
}