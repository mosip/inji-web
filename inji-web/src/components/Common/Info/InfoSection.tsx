import React, {Fragment} from "react";
import {InfoSectionStyles} from "./InfoSectionStyles";
import LayoutStyles from "../LayoutStyles";

type InfoSectionProps =
    {
        title?: string,
        message?: string
        icon?: React.ReactElement
        mobileAction?: React.ReactElement
        testId: string
    }

export function InfoSection({
                                title,
                                message,
                                icon,
                                testId,
                                mobileAction
                            }: Readonly<InfoSectionProps>): React.ReactElement {
    return <div
        data-testid={`${testId}-container`}
        className={LayoutStyles.sectionWithWhiteBg}
    >
        {icon && <Fragment>{icon}</Fragment>}
        <div className={InfoSectionStyles.content}>
            {title && <h2
                data-testid={`${testId}-title`}
                className={InfoSectionStyles.title}
            >
                {title}
            </h2>}
            {message && <span className={InfoSectionStyles.message} data-testid={`${testId}-message`}>{message}</span>}
        </div>
        {mobileAction &&
            <div className={InfoSectionStyles.action}>
                {mobileAction}
            </div>
        }
    </div>;
}