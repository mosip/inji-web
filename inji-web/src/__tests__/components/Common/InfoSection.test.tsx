import React from 'react';
import {render, screen} from '@testing-library/react';
import {InfoSection} from '../../../components/Common/Info/InfoSection';

function MockIcon() {
    return <svg data-testid="mock-icon"/>;
}

describe('InfoSection Component', () => {
    it('should match snapshot with all props', () => {
        const {container} = render(
            <InfoSection
                title="Test Title"
                message="Test Action Text"
                icon={<MockIcon/>}
                testId="test-info-section"
            />
        );

        expect(container).toMatchSnapshot();
    });

    it('should match snapshot with minimal props', () => {
        const {container} = render(
            <InfoSection message="Test Action Text Only" testId="test-info-section"/>
        );

        expect(container).toMatchSnapshot();
    });

    it('should render with all props', () => {
        render(
            <InfoSection
                title="Test Title"
                message="Test Action Text"
                icon={<MockIcon/>}
                testId="test-info-section"
            />
        );

        expect(screen.getByText("Test Title")).toBeInTheDocument();
        expect(screen.getByText('Test Action Text')).toBeInTheDocument();
        expect(screen.getByTestId('test-info-section-title')).toHaveTextContent('Test Title');
        expect(screen.getByTestId("test-info-section-message")).toHaveTextContent('Test Action Text');
        expect(document.querySelector('svg')).toBeInTheDocument();
    });

    it('should render only with actionText', () => {
        render(
            <InfoSection message="Only Action Text" testId="test-info-section"/>
        );

        expect(screen.queryByTestId('test-info-section-title')).not.toBeInTheDocument();
        expect(screen.getByText('Only Action Text')).toBeInTheDocument();
        expect(screen.getByTestId("test-info-section-message")).toHaveTextContent('Only Action Text');
        expect(document.querySelector('svg')).not.toBeInTheDocument();
    });

    it('should render only with title', () => {
        render(
            <InfoSection title="Only Title" testId="test-info-section"/>
        );

        expect(screen.getByTestId('test-info-section-title')).toBeInTheDocument();
        expect(screen.getByTestId('test-info-section-title')).toHaveTextContent('Only Title');
        expect(screen.queryByText('Test Action Text')).not.toBeInTheDocument();
        expect(document.querySelector('svg')).not.toBeInTheDocument();
    });

    it('should render only with icon', () => {
        render(
            <InfoSection icon={<MockIcon/>} testId="test-info-section"/>
        );

        expect(screen.queryByTestId('test-info-section-title')).not.toBeInTheDocument();
        expect(document.querySelector('svg')).toBeInTheDocument();
    });
});