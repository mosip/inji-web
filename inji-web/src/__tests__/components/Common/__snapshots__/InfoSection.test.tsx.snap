// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoSection Component should match snapshot with all props 1`] = `
<div>
  <div
    class="bg-white rounded-lg shadow-iw-layout p-6 sm:p-5 md:p-10 flex flex-col items-center justify-center min-h-[400px] sm:min-h-[450px] mt-4 sm:mt-8"
    data-testid="test-info-section-container"
  >
    <svg
      data-testid="mock-icon"
    />
    <div
      class="inline-flex flex-col items-center max-w-[370px]"
    >
      <h2
        class="text-xl text-center sm:text-2xl font-medium text-gray-800 mb-2"
        data-testid="test-info-section-title"
      >
        Test Title
      </h2>
      <span
        class="text-[var(--iw-color-info)] font-normal text-sm break-words justify-center text-center"
        data-testid="test-info-section-message"
      >
        Test Action Text
      </span>
    </div>
  </div>
</div>
`;

exports[`InfoSection Component should match snapshot with minimal props 1`] = `
<div>
  <div
    class="bg-white rounded-lg shadow-iw-layout p-6 sm:p-5 md:p-10 flex flex-col items-center justify-center min-h-[400px] sm:min-h-[450px] mt-4 sm:mt-8"
    data-testid="test-info-section-container"
  >
    <div
      class="inline-flex flex-col items-center max-w-[370px]"
    >
      <span
        class="text-[var(--iw-color-info)] font-normal text-sm break-words justify-center text-center"
        data-testid="test-info-section-message"
      >
        Test Action Text Only
      </span>
    </div>
  </div>
</div>
`;
